sdk/typescript/test/unit/utils/iota-types.test.ts

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Mysten Labs, Inc.
+// Modifications Copyright (c) 2024 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+import { describe, expect, it } from 'vitest';
+import { normalizeIotaAddress } from '../../../src/utils/iota-types';
+
+describe('normalizeIotaAddress', () => {
+    it('should convert address to lowercase', () => {
+        expect(normalizeIotaAddress('0xABC123')).toBe(
+            '0x0000000000000000000000000000000000000000000000000000000000abc123',
+        );
+    });
+
+    it('should remove whitespaces', () => {
+        expect(normalizeIotaAddress(' 0xab  c1 23 ')).toBe(
```

Comment:
Accepting such strings doesn't look like a good idea to me, if it has spaces in between then it's probably some other value and not an address and we should return an error to be save

Reply:
if we go with my suggestion, lets update all these tests so they consider validation on/off